### PR TITLE
Add rel.project operator to sets

### PR DESCRIPTION
**Pull request by mudathirmahgoub**

### CODE REVIEW

File: test/regress/cli/regress1/sets/relation_project2.smt2 [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+(set-option :fmf-bound true)
+(set-option :uf-lazy-ll true)
+
+(declare-fun A () (Relation String String))
+(declare-fun B () (Relation String String))
+
+(assert (distinct A (as set.empty (Relation String String))))
+(assert (= B ((_ rel.project 1 0) A)))
+(assert (set.member (tuple "y" "x") B))
+(check-sat)
```
Comment (mudathirmahgoub) [inline test/regress/cli/regress1/sets/relation_project2.smt2:12]: cvc5 returns unknown answer for the the bag version of this example. 
That is why it is not included in this PR

Comment (mudathirmahgoub) [reply test/regress/cli/regress1/sets/relation_project2.smt2:10]: Removing (distinct A bag.empty) was enough for cvc5 to answer sat.

File: src/theory/sets/theory_sets_type_rules.cpp [hunk 2]
```diff
@@ -632,7 +633,7 @@ TypeNode RelationAggregateTypeRule::computeType(NodeManager* nm,
     if (!setType.isSet())
     {
       std::stringstream ss;
-      ss << "RELATION_PROJECT operator expects a table. Found '" << n[2]
+      ss << "RELATION_AGGREGATE operator expects a table. Found '" << n[2]
```
Comment (ajreynol) [inline src/theory/sets/theory_sets_type_rules.cpp:636]:
```suggestion
      ss << "RELATION_AGGREGATE operator expects a set. Found '" << n[2]
```

File: src/theory/sets/theory_sets_type_rules.cpp [hunk 3]
```diff
@@ -641,7 +642,7 @@ TypeNode RelationAggregateTypeRule::computeType(NodeManager* nm,
     if (!tupleType.isTuple())
     {
       std::stringstream ss;
-      ss << "TABLE_PROJECT operator expects a table. Found '" << n[2]
+      ss << "RELATION_AGGREGATE operator expects a table. Found '" << n[2]
```
Comment (ajreynol) [inline src/theory/sets/theory_sets_type_rules.cpp:645]:
```suggestion
      ss << "RELATION_AGGREGATE operator expects a relation. Found '" << n[2]
```

File: src/theory/sets/theory_sets_type_rules.cpp [hunk 4]
```diff
@@ -680,6 +681,63 @@ TypeNode RelationAggregateTypeRule::computeType(NodeManager* nm,
...
+      ss << "RELATION_PROJECT operator expects a relation. Found '" << n[0]
+         << "' of type '" << setType << "'.";
+      throw TypeCheckingExceptionPrivate(n, ss.str());
+    }
+
+    TypeNode tupleType = setType.getSetElementType();
+    if (!tupleType.isTuple())
+    {
+      std::stringstream ss;
+      ss << "RELATION_PROJECT operator expects a table. Found '" << n[0]
```
Comment (ajreynol) [inline src/theory/sets/theory_sets_type_rules.cpp:715]:
```suggestion
      ss << "RELATION_PROJECT operator expects a relation. Found '" << n[0]
```

**Pull request merged**